### PR TITLE
[5.0] Smart Search: Allow to hide archived content

### PR DESCRIPTION
**Pull request by Hackwar**
Pull Request for Issue #41339.

### Summary of Changes
The old com_search had an option to search both published and archived content. This PR adds that option to the core Smart Search plugins as well. By default archived content is searched, since that was the behavior until now.

### Testing Instructions
1. Install the testing sampledata
2. Set an article to archived
3. Search for that article in smart search
4. Switch the option in the article finder plugin to no.
5. Edit the archived article and change something small in it.
6. Search for the article again
7. Change the state of the article back to published.
8. Search for the article again.

### Actual result BEFORE applying this Pull Request
The article is always returned as a search result.

### Expected result AFTER applying this Pull Request
When at step 6, the article does not show up in the search results anymore.
At step 8, the article does show up again.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: plugins/finder/categories/categories.xml [hunk 1]
```diff
@@ -21,6 +21,18 @@
...
 			<fieldset name="basic" addfieldprefix="Joomla\Component\Finder\Administrator\Field">
+				<field
+					name="search_archived"
+					type="radio"
+					layout="joomla.form.field.radio.switcher"
+					label="PLG_FINDER_CATEGORIES_FIELD_SEARCH_ARCHIVED_LABEL"
+					description="PLG_FINDER_CATEGORIES_FIELD_SEARCH_ARCHIVED_DESC"
+					default="1"
+					filter="integer"
+				>
```
Comment (Quy) [inline plugins/finder/categories/categories.xml]:
```suggestion
					>
```

File: plugins/finder/newsfeeds/newsfeeds.xml [hunk 2]
```diff
@@ -21,6 +21,18 @@
 	<config>
 		<fields name="params">
 			<fieldset name="basic" addfieldprefix="Joomla\Component\Finder\Administrator\Field">
+				<field
+					name="search_archived"
+					type="radio"
+					layout="joomla.form.field.radio.switcher"
+					label="PLG_FINDER_NEWFEEDS_FIELD_SEARCH_ARCHIVED_LABEL"
```
Comment (Quy) [inline plugins/finder/newsfeeds/newsfeeds.xml]:
```suggestion
					label="PLG_FINDER_NEWSFEEDS_FIELD_SEARCH_ARCHIVED_LABEL"
```

**Comment by HLeithner** [comment]
Nice feature, allowing to not index any article would be nice too.

**Comment by Hackwar** [comment]
If you don't want to index any articles at all, you can just switch off the content finder plugin. 😉 But I understand that you want an option to not index a content item in the first place. Will see what I can do.

**Comment by HLeithner** [comment]
yeah, my wording wasn't specific enough :-(

File: plugins/finder/contacts/contacts.xml [hunk 3]
```diff
@@ -21,6 +21,18 @@
...
 			<fieldset name="basic" addfieldprefix="Joomla\Component\Finder\Administrator\Field">
+				<field
+					name="search_archived"
+					type="radio"
+					layout="joomla.form.field.radio.switcher"
+					label="PLG_FINDER_CONTACTS_FIELD_SEARCH_ARCHIVED_LABEL"
+					description="PLG_FINDER_CONTACTS_FIELD_SEARCH_ARCHIVED_DESC"
+					default="1"
+					filter="integer"
+				>
```
Comment (Quy) [inline plugins/finder/contacts/contacts.xml]:
```suggestion
					>
```

File: plugins/finder/content/content.xml [hunk 4]
```diff
@@ -21,6 +21,18 @@
...
 			<fieldset name="basic" addfieldprefix="Joomla\Component\Finder\Administrator\Field">
+				<field
+					name="search_archived"
+					type="radio"
+					layout="joomla.form.field.radio.switcher"
+					label="PLG_FINDER_CONTENT_FIELD_SEARCH_ARCHIVED_LABEL"
+					description="PLG_FINDER_CONTENT_FIELD_SEARCH_ARCHIVED_DESC"
+					default="1"
+					filter="integer"
+				>
```
Comment (Quy) [inline plugins/finder/content/content.xml]:
```suggestion
					>
```

File: plugins/finder/newsfeeds/newsfeeds.xml [hunk 5]
```diff
@@ -21,6 +21,18 @@
...
 			<fieldset name="basic" addfieldprefix="Joomla\Component\Finder\Administrator\Field">
+				<field
+					name="search_archived"
+					type="radio"
+					layout="joomla.form.field.radio.switcher"
+					label="PLG_FINDER_NEWSFEEDS_FIELD_SEARCH_ARCHIVED_LABEL"
+					description="PLG_FINDER_NEWSFEEDS_FIELD_SEARCH_ARCHIVED_DESC"
+					default="1"
+					filter="integer"
+				>
```
Comment (Quy) [inline plugins/finder/newsfeeds/newsfeeds.xml]:
```suggestion
					>
```

File: plugins/finder/tags/tags.xml [hunk 6]
```diff
@@ -21,6 +21,18 @@
...
 			<fieldset name="basic" addfieldprefix="Joomla\Component\Finder\Administrator\Field">
+				<field
+					name="search_archived"
+					type="radio"
+					layout="joomla.form.field.radio.switcher"
+					label="PLG_FINDER_TAGS_FIELD_SEARCH_ARCHIVED_LABEL"
+					description="PLG_FINDER_TAGS_FIELD_SEARCH_ARCHIVED_DESC"
+					default="1"
+					filter="integer"
+				>
```
Comment (Quy) [inline plugins/finder/tags/tags.xml]:
```suggestion
					>
```

**Comment by HLeithner** [comment]
documentation is needed here for 3rd party developer and also for users....

**Comment by HLeithner** [comment]
thanks

**Comment by 1inkin** [comment]
Is this feature still planned for inclusion in the Joomla distribution?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41504">issues.joomla.org/tracker/joomla-cms/41504</a>.</sub>

**Comment by alikon** [comment]
it is already  in 5

**Comment by 1inkin** [comment]
I found the switch in the plugin settings. Thank you. very useful feature. It would be nice to add a switch to the article parameters to disable indexing of some articles.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41504">issues.joomla.org/tracker/joomla-cms/41504</a>.</sub>

**Comment by peterhulst** [comment]
In Joomla 5 the Smart Search plugins have the (new) selection: **Search Archived Content**: Yes/No.
It would make sense to extend this selection in the separate plugins with: Yes/No/**Global**.
In the Options for the Smart Search component the global value can be selected: **Search Archived Content (global)**: Yes/No

Regards,
Peter<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41504">issues.joomla.org/tracker/joomla-cms/41504</a>.</sub>

**Pull request merged**